lib/dams/dams_object.rb

```diff
@@ -391,6 +382,23 @@ def insertFileFields( solr_doc, cid, files )
           end
 	    }
 	  end
+      def fulltext( object, file, cid )
+        mime_type = file.mimeType.first.to_s.gsub(/;.*/,"")
+        if ["application/pdf", "text/html", "text/plain"].include?(mime_type)
+          fid = (cid != nil) ? "_#{cid}_#{file.id}" : "_" + file.id
+          fid = "fulltext#{fid}" if mime_type == "application/pdf"
```

Comment:
HTML and plain text can simply be retrieved as-is -- PDFs need to have the fulltext prefix to trigger the DAMS REST API text extraction service with Tika.